src/auth/bearer/fake/fake.cc

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2014, Treehouse Networks Ltd. New Zealand
```

Comment:
I disagree with your unsubstantiated analysis. I believe your opinion on how copyright works is flat wrong. Copyright law does not have a concept of a "file". Modern copy rights do not magically disappear (or appear) when statements are moved. IANAL, but I can provide references to substantiate my opinion:

> [Software Freedom Law Center](https://softwarefreedom.org/resources/2012/ManagingCopyrightInformation.html): a requirement to “preserve” or “reproduce” a developer’s copyright notice does not necessarily require that the notice be kept in exactly the same place it started; it’s usually acceptable to move notices from individual source files to a central attribution file, for example.

> [Larry Rosen](https://issues.apache.org/jira/browse/LEGAL-124?focusedCommentId=13205693&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13205693): Additional license headers can either be in the source file itself or in a separate LICENSE file. ...  I'd suggest that we move such additional information to a separate LICENSE file.

> [Craig L Russell](https://issues.apache.org/jira/browse/LEGAL-124?focusedCommentId=13207071&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13207071): The [Apache Foundation] decision in 2006 was to move copyright notices from source files, leaving a License Header behind.

> [Software Freedom Law Center](https://softwarefreedom.org/resources/2012/ManagingCopyrightInformation.html): Since copyright notices are not mandatory, there is generally no harm in under-using them, particularly if authorship is recorded in VCS logs. Over-use, however, can cause problems.

Every project has its nuances and custom policies. It is easy to find recommendations to preserve 3rd party notices in some special cases, as defined by that project. I do not think this file is such a special case from the Squid Project point of view. It is hardly creative enough to deserve a copyright, and certainly not special enough to deserve an embedded copyright statement.